pydis_site/apps/home/views/home.py

```diff
@@ -0,0 +1,110 @@
+import requests
+from django.shortcuts import render
+from django.utils import timezone
+from django.views import View
+
+from pydis_site.apps.home.models import RepositoryMetadata
+
+
+class HomeView(View):
+    """The view"""
+
+    github_api = "https://api.github.com/users/python-discord/repos"
+
+    # Which of our GitHub repos should be displayed on the front page, and in which order?
+    repos = [
+        "python-discord/site",
+        "python-discord/bot",
+        "python-discord/snekbox",
+        "python-discord/seasonalbot",
+        "python-discord/django-simple-bulma",
+        "python-discord/django-crispy-bulma",
+    ]
+
+    def _get_api_data(self):
+        """Call the GitHub API and get information about our repos."""
+
+        repo_dict = {repo_name: {} for repo_name in self.repos}
+
+        # Fetch the data from the GitHub API
+        api_data = requests.get(self.github_api)
+        api_data = api_data.json()
+
+        # Process the API data into our dict
+        for repo in api_data:
+            full_name = repo["full_name"]
+
+            if full_name in self.repos:
+                repo_dict[full_name] = {
+                    "full_name": repo["full_name"],
+                    "description": repo["description"],
+                    "language": repo["language"],
+                    "forks_count": repo["forks_count"],
+                    "stargazers_count": repo["stargazers_count"],
+                }
+        return repo_dict
+
+    def _get_repo_data(self):
+        """Build a list of RepositoryMetadata objects that we can use to populate the front page."""
+
+        # Try to get site data from the cache
+        try:
+            repo_data = RepositoryMetadata.objects.get(repo_name="python-discord/site")
+
+            # If the data is older than 2 minutes, we should refresh it.
+            if (timezone.now() - repo_data.last_updated).seconds > 120:
```

Comment:
isn't 2 minutes a bit often though? that said, maybe this is also better as a class constant? `repository_cache_ttl` or something like that

Reply:
Okay so, it looks like we get 60 of these per hour with unauthenticated access. This increases to 5000 if we do bother to auth, so if we needed more, it'd be easy enough to get them, but I'd rather not add authentication if we don't need to.

For 60 an hour, that means that this would exhaust 50% of the rate limit capacity, which, yeah, is maybe a little high when the only advantage is quicker tracking of new stars or whatever.

I'm gonna set it to 10 minutes for now, how does that sound?